app/components/pipeline/Header/index.js

```diff
@@ -189,7 +174,7 @@ class Header extends React.Component {
     if (this.props.pipeline.description) {
       return <Emojify text={this.props.pipeline.description} />;
     } else {
-      return this.repositoryUrl();
+      return this.props.pipeline.repository.url;
```

Comment:
All good! As soon as I spotted that Javascript I was like "We should have that in Ruby already" but didn't get a chance to fully check it out.

These are the GraphQL models for it:

```ruby
module API::Graph::Types
  RepositoryType = GraphQL::ObjectType.define do
    name "Repository"
    description "A repository associated with a pipeline"

    field :url do
      type !types.String
      description "The URL for this repository"
      resolve -> (project, arguments, context) {
        project.repository
      }
    end

    field :provider, -> { API::Graph::Interfaces::RepositoryProviderInterface } do
      resolve -> (project, arguments, context) do
        Project::Provider.new(project)
      end
    end
  end
end
```

v.s.

```ruby
module API::Graph::Interfaces
  RepositoryProviderInterface = GraphQL::InterfaceType.define do
    name "RepositoryProvider"

    field :name, !types.String do
      resolve -> (provider, arguments, context) do
        provider.project.provider.name
      end
    end

    field :url, types.String do
      resolve -> (provider, arguments, context) do
        provider.project.provider.project_url(provider.project)
      end
    end
  end
end
```

`project.repository` is just the attribute from the ActiveRecord model.

`provider.project.provider.project_url` is the `#project_url` method of all the `Project::Provider::*` subclasses